src/translate/function.rs

```diff
@@ -78,6 +78,34 @@ pub fn translate_function(sql_function: &SqlFunction) -> Result<Expr> {
 
             Ok(Expr::Function(Box::new(Function::Right { expr, size })))
         }
+        "LTRIM" => {
+            let expr;
+            let chars;
+            if let Ok(()) = check_len(name.clone(), args.len(), 1) {
+                expr = translate_expr(args[0])?;
+                chars = Expr::Literal(AstLiteral::Null);
+            } else if let Ok(()) = check_len(name, args.len(), 2) {
+                expr = translate_expr(args[0])?;
+                chars = translate_expr(args[1])?;
+            } else {
+                return Err(TranslateError::UnsupportedExpr(String::from("LTRIM")).into());
+            }
+            Ok(Expr::Function(Box::new(Function::Ltrim { expr, chars })))
+        }
```

Comment:
Here, `match` would be good to simplify codes and reduce redundant `check_len` calls.

fyi. below codes are two different examples, I think the second one is better in this case.
1.
```rust
"LTRIM" => match args.len() {                                                     
    1 => translate_expr(args[0])                                                  
        .map(|expr| Function::Ltrim { expr, chars: None })                        
        .map(Box::new)                                                            
        .map(Expr::Function),                                                     
    2 => translate_expr(args[0])                                                  
        .and_then(|expr| translate_expr(args[1]).map(|chars| (expr, Some(chars))))
        .map(|(expr, chars)| Function::Ltrim { expr, chars })                     
        .map(Box::new)                                                            
        .map(Expr::Function),                                                     
    _ => Err(TranslateError::UnsupportedExpr(String::from("LTRIM")).into()),      
},                                                                                
```

2.
```rust
"LTRIM" => {                                                                
    let (expr, chars) = match args.len() {                                  
        1 => (translate_expr(args[0])?, None)                               
        2 => (translate_expr(args[0])?, translate_expr(args[1]).map(Some)?),
        n => {                                                              
            return Err(TranslateError::FunctionArgsLengthRangeNotMatching { 
                name: "LTRIM".to_owned(),                                   
                min: 1,                                                     
                max: 2,                                                     
                found: n,                                                   
            });                                                             
        }                                                                   
    };                                                                      
                                                                            
    Ok(Expr::Function(Box::new(Function::Ltrim { expr, chars })))           
}                                                                           
```

And.. you don't need to hesitate about adding errors, please feel free to add errors if you feel that there is a new case which existing errors don't cover.

In your case the existing one `FunctionArgsLengthNotMatching` does not fit to your case because `LTRIM` and `RTRIM` accepts one or two params. Then, you can add a new one something like this.
```rust
#[error("function args.length not matching: {name}, expected: {expected}, found: {found}")]    
FunctionArgsLengthNotMatching {                                                                
    name: String,                                                                              
    expected: usize,                                                                           
    found: usize,                                                                              
},                                                                                             
                                                                                               
#[error("function args.length range not matching: {name}, range: {min}-{max}, found: {found}")]
FunctionArgsLengthRangeNotMatching {                                                           
    name: String,                                                                              
    min: usize,                                                                                
    max: usize,                                                                                
    found: usize,                                                                              
},                                                                                             
```

Reply:
@devgony
Thanks to @No-YE, he added `check_len_range` closure to `translate/function` module.
ref. https://github.com/gluesql/gluesql/pull/311

It would good to apply in here :)
https://github.com/gluesql/gluesql/blob/83db4959ecf884c0fcfc5bf7b129add3a44760b5/src/translate/function.rs#L40-L52